src/components/views/create/NameInput.tsx

```diff
@@ -1,25 +1,62 @@
-import React from 'react';
-import { H5 } from '@giveth/ui-design-system';
-import { Regular_Input } from '@/components/styled-components/Input';
-import { InputContainer, Label } from './Create.sc';
+import React, { useState } from 'react';
+import { H5, neutralColors } from '@giveth/ui-design-system';
 import styled from 'styled-components';
+import {
+	InputContainer,
+	InputErrorMessage,
+	InputWithError,
+	Label,
+} from './Create.sc';
+import { FlexCenter } from '@/components/styled-components/Grid';
+import { ECreateErrFields } from '@/components/views/create/CreateIndex';
 
-const NameInput = (props: any) => {
-	const { setValue } = props;
+const NameInput = (props: {
+	value: string;
+	setValue: (e: string) => void;
+	error: string;
+}) => {
+	const { setValue, value, error } = props;
+	const [characterLength, setCharacterLength] = useState(0);
+	const maxLength = 55;
 	return (
 		<>
-			<H5>Name of your Project</H5>
+			<H5 id={ECreateErrFields.NAME}>Name of your Project</H5>
 			<InputContainer>
 				<Label>Project name</Label>
-				<Input
-					placeholder='My First Project'
-					onChange={e => setValue(e.target.value)}
-				/>
+				<div style={{ position: 'relative' }}>
+					<InputWithError
+						placeholder='My First Project'
+						onChange={e => {
+							setValue(e.target.value);
+							setCharacterLength(e.target.value.length);
+						}}
+						value={value}
+						maxLength={maxLength}
+						error={!!error}
+					/>
+					<CharLength>
+						{characterLength}/{maxLength}
+					</CharLength>
+				</div>
+				<InputErrorMessage>{error || null}</InputErrorMessage>
 			</InputContainer>
 		</>
 	);
 };
 
-const Input = styled(Regular_Input)``;
+const CharLength = styled(FlexCenter)`
+	font-size: 12px;
+	background: ${neutralColors.gray[300]};
+	color: ${neutralColors.gray[700]};
+	font-weight: 500;
+	border-radius: 64px;
+	width: 52px;
+	height: 30px;
+	position: absolute;
+	right: 16px;
+	top: 0;
+	bottom: 0;
+	margin: auto 0;
+`;
```

Comment:
Do we have a typography component for it?

Reply:
We can use one styled component for each one